posthog/session_recordings/queries/session_recording_list_from_filters.py

```diff
@@ -258,39 +235,8 @@ def _where_predicates(self) -> Union[ast.And, ast.Or]:
 
         return ast.And(exprs=exprs)
 
-    def _having_predicates(self) -> ast.And | Constant:
-        exprs: list[ast.Expr] = []
-
-        if self._filter.recording_duration_filter:
-            op = (
-                ast.CompareOperationOp.GtEq
-                if self._filter.recording_duration_filter.operator == "gt"
-                or self._filter.recording_duration_filter.operator == "gte"
-                else ast.CompareOperationOp.LtEq
-            )
-            exprs.append(
-                ast.CompareOperation(
-                    op=op,
-                    left=ast.Field(chain=[self._filter.duration_type_filter]),
-                    right=ast.Constant(value=self._filter.recording_duration_filter.value),
-                ),
-            )
-
-        if self._filter.snapshot_source_filter:
-            op = (
-                ast.CompareOperationOp.In
-                if self._filter.snapshot_source_filter.operator == "exact"
-                else ast.CompareOperationOp.NotIn
-            )
-            exprs.append(
-                ast.CompareOperation(
-                    op=op,
-                    left=ast.Call(name="argMinMerge", args=[ast.Field(chain=["s", "snapshot_source"])]),
-                    right=ast.Constant(value=self._filter.snapshot_source_filter.value),
-                ),
-            )
-
-        return ast.And(exprs=exprs) if exprs else ast.Constant(value=True)
```

Comment:
Being able to get rid of all this kind of custom filtering code is the main intention behind this PR. Everything is now handled inside the single `property_to_expr` method